src/zarr/core/group.py

```diff
@@ -355,10 +355,10 @@ def to_buffer_dict(self, prototype: BufferPrototype) -> dict[str, Buffer]:
                 assert isinstance(consolidated_metadata, dict)
                 for k, v in consolidated_metadata.items():
                     attrs = v.pop("attributes", None)
-                    d[f"{k}/{ZATTRS_JSON}"] = attrs
+                    d[f"{k}/{ZATTRS_JSON}"] = _replace_special_floats(attrs)
```

Comment:
@mpiannucci was this particular change important? Given that the "special floats" logic is very narrowly scoped to array metadata, I don't think we want to apply it to the attributes.

I'm just seeing this now as I adapt #2874 to the latest changes.